### ISSUE TITLE
QtFirebase hard for use in project

### ISSUE THREAD

**Issue by TechComet** [Larpon/QtFirebase#48]
QtFirebase hard for use in project

**Comment by greenfield932** [comment]
What is the problem? Did you take a look at the sample app?
https://github.com/Larpon/QtFirebaseExample

**Comment by greenfield932** [comment]
By the way you are welcome to make it better)

**Comment by Larpon** [comment]
I agree that the setup process is not optimal. IMHO the problem itself lies in how Qt and QMake behaves when setting up 3rd party projects and cross-compilation (it's not straight forward to handle `gradle` and `Info.plist` dependencies etc. via QMake). Also the Firebase setup on the different target platforms (Android / iOS) is a little suboptimal.

I'm hoping to get enough time on my hands to try my luck with `qbs` soon (others are welcome to contribute the foundation for this!). I hope we can skip a lot of the setup steps - and preferably end up just having people provide:
`google-services.json`
`GoogleService-Info.plist`
`QTFIREBASE_CONFIG`

I've also thought about a small GUI application for the actual setup but that's undecided and only "nice to have" at this point

**Comment by Larpon** [comment]
@rmallah - you're welcome - and thanks for clearifying.

I didn't know about the `gradle.properties` and `local.properties` shouldn't be checked in to a VCS.

I'll correct the paths you mention in the documentation - they seem to be wrong.

I often emphasize that people should see how [QtFirebaseExample](https://github.com/Larpon/QtFirebaseExample) is setup - as the master branch will *usually* always be working with the master branch of the QtFirebase project. Some of your points are in the Example app.

I will add some better details to `SETUP.md` about the gradle stuff

**Comment by ahmetcumhurarslan** [comment]
> I agree that the setup process is not optimal. IMHO the most problems lie in how Qt and QMake behaves when setting up 3rd party projects and cross-compilation (it's not straight forward to handle `gradle` and `Info.plist` dependencies / content etc. dynamically via QMake). Also the Firebase setup on the different target platforms (Android / iOS) is a little suboptimal in some cases (e.g. the Cloud Messaging where you have to activate certificates etc.).
> 
> I'm hoping to get enough time on my hands to try my luck with `qbs` soon (others are welcome to contribute the foundation for this!). I hope we can skip a lot of the setup steps - and preferably end up just having people provide:
> `google-services.json`
> `GoogleService-Info.plist`
> `QTFIREBASE_CONFIG`
> 
> I've also thought about a small GUI application for the actual setup but that's undecided and only "nice to have" at this point.
> 
> And as @greenfield932 mentioned: We gladly accept any pull requests with improvements!

a while ego i wrote a GUI which downloads required files, modifies them  and creates a template start point using qt firebase. It creates both android and ios according to user specifications. however the structure is changed and as i stated [here](https://github.com/Larpon/QtFirebase/issues/92#issuecomment-582897386) i could not compile the ios side of the applications. if you decide to go through this "GUI" way of improvement maybe i can help.

**Comment by davidzwa** [comment]
@ahmetcumhurarslan I am interested in pulling your GUI into this repo, or collab to make this work. I see why and how this is essential and we have the same necessity for using QtFirebase.

**Comment by Larpon** [comment]
See also efforts and discussion in https://github.com/Larpon/QtFirebaseExample/issues/32
